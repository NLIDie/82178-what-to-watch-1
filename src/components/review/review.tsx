import * as React from "react";

// Utils
import {getDate} from "_Utils/get-date";

// Types
import {TProps} from "./review.types";

export function Review(props: TProps): JSX.Element {
  const {userName, rating, comment, date} = props;

  const newDate = getDate(new Date(date));

  return (
    <div className="review">
      <blockquote className="review__quote">
        <p className="review__text">{comment}</p>

        <footer className="review__details">
          <cite className="review__author">{userName}</cite>
          <time className="review__date" dateTime={date}>
            {newDate.month} {newDate.day}, {newDate.year}
          </time>
        </footer>
      </blockquote>

      <div className="review__rating">{rating}</div>
    </div>
  );
}
