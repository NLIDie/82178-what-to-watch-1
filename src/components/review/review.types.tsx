export type TProps = {
  userName: string;
  rating: number;
  comment: string;
  date: string;
};
