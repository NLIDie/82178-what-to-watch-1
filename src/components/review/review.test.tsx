import * as React from "react";
import {create} from "react-test-renderer";

// Components
import {Review} from "./review";

describe(`Components/Review`, () => {
  it(`Correct renderer the component`, () => {
    const tree = create(
        <Review
          userName="Sophie"
          rating={2}
          date="2019-06-03T17:29:12.632Z"
          comment="It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie."
        />
    );

    expect(tree).toMatchSnapshot();
  });
});
