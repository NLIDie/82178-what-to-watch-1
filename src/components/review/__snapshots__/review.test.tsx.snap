// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Review Correct renderer the component 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    >
      It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.
    </p>
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      >
        Sophie
      </cite>
      <time
        className="review__date"
        dateTime="2019-06-03T17:29:12.632Z"
      >
        June
         
        3
        , 
        2019
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  >
    2
  </div>
</div>
`;
